qa/vagrant/src/main/java/org/elasticsearch/packaging/util/Archives.java

```diff
@@ -0,0 +1,239 @@
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.packaging.util;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.elasticsearch.packaging.util.FileMatcher.Fileness.Directory;
+import static org.elasticsearch.packaging.util.FileMatcher.Fileness.File;
+import static org.elasticsearch.packaging.util.FileMatcher.file;
+import static org.elasticsearch.packaging.util.FileMatcher.p644;
+import static org.elasticsearch.packaging.util.FileMatcher.p660;
+import static org.elasticsearch.packaging.util.FileMatcher.p755;
+import static org.elasticsearch.packaging.util.FileUtils.getCurrentVersion;
+import static org.elasticsearch.packaging.util.FileUtils.getDefaultArchiveInstallPath;
+import static org.elasticsearch.packaging.util.FileUtils.getPackagingArchivesDir;
+import static org.elasticsearch.packaging.util.FileUtils.lsGlob;
+
+import static org.elasticsearch.packaging.util.FileUtils.mv;
+import static org.elasticsearch.packaging.util.Platforms.isDPKG;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.collection.IsEmptyCollection.empty;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+
+/**
+ * Installation and verification logic for archive distributions
+ */
+public class Archives {
+
+    public static Installation installArchive(Distribution distribution) {
+        return installArchive(distribution, getDefaultArchiveInstallPath(), getCurrentVersion());
+    }
+
+    public static Installation installArchive(Distribution distribution, Path fullInstallPath, String version) {
+        final Shell sh = new Shell();
+
+        final Path distributionFile = getPackagingArchivesDir().resolve(distribution.filename(version));
+        final Path baseInstallPath = fullInstallPath.getParent();
+        final Path extractedPath = baseInstallPath.resolve("elasticsearch-" + version);
+
+        assertThat("distribution file must exist", Files.exists(distributionFile), is(true));
+        assertThat("elasticsearch must not already be installed", lsGlob(baseInstallPath, "elasticsearch*"), empty());
+
+        if (distribution.packaging == Distribution.Packaging.TAR) {
```

Comment:
I wonder if this kind of thing should be moved to `ArchiveTestCase` and you subclass it for each time.

Reply:
Like you move the entire `installArchive` method to the abstract superclass.

Reply:
I think it is fine to do it as you are doing it. We could change it to my way later on if we figure it is better. I don't think it is a blocker.

I just look at your way and think "this made a ton of sense in bash scripts but I want to OO all the things in Java because I'm a Java developer." I admit that is a bad habit some times. I don't think it'd be too weird to keep the install logic in the superclass but you certainly are right that there aren't that many variants.

Reply:
>this made a ton of sense in bash scripts but I want to OO all the things in Java because I'm a Java developer

Yeah this was exactly what I did at first and then walked it back to here since it seemed simpler :)

I'll take a look at the other places in Archives' methods where this kind of stuff happens and see if I can find a happy medium

Reply:
I tried a couple variations of what we discussed. I still feel like the benefit of being able to answer the question "how are the archive distros installed" by looking in one place outweighs that place being a little gross - not in general, but in this instance because we don't expect to have many, if any additional cases. 

I agree that if it does change it would be better to do something like your suggestion. I'll also keep thinking about other ways to do it as I revise this